src/hooks/mobile.tsx

```diff
@@ -12,8 +12,14 @@ const enum MobileSearchParams {
 export const useMobileAppLogin = () => {
   const [searchParams] = useSearchParams();
   const challenge = searchParams.get(MobileSearchParams.CHALLENGE);
+  if (challenge !== undefined && challenge !== null) {
+    return {
+      isMobile: true as const,
+      challenge,
+    };
+  }
   return {
-    isMobile: challenge !== undefined && challenge !== null,
+    isMobile: false as const,
     challenge,
   };
```

Comment:
Here I changed the code so the return type is a discriminated union that allows the challenge to be linked to the value of the `isMobile` boolean for type narrowing.